rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java

```diff
@@ -1842,6 +1852,13 @@ private Optional<String> getRevisionFromCalledWorkflow(String calledWorkflow) {
         }
     }
 
+    public static Map<String, String> setSubmissionModeToGenericInfo(Map<String, String> genericInformation) {
+        Map<String, String> newGenericInformation = genericInformation == null ? new LinkedHashMap<>()
+                                                                               : new LinkedHashMap<>(genericInformation);
+        newGenericInformation.put(SUBMISSION_MODE, SUBMISSION_MODE_SCHEDULER_API);
```

Comment:
could you check if SUBMISSION_MODE is not already in the generic info, if yes, then don't set it